src/sentry/integrations/slack/utils/channel.py

```diff
@@ -93,8 +93,9 @@ def validate_channel_id(name: str, integration_id: Optional[int], input_channel_
         raise IntegrationError("Bad slack channel list response.")
 
     stripped_channel_name = strip_channel_name(name)
-    if not stripped_channel_name == results["channel"]["name"]:
-        channel_name = results["channel"]["name"]
+    results_channel_name = results.get("channel", {}).get("name")
+    if not stripped_channel_name == results_channel_name:
+        channel_name = results_channel_name
         raise ValidationError(
             f"Received channel name {channel_name} does not match inputted channel name {stripped_channel_name}."
```

Comment:
Would this be "... channel name None does not ..." if channel does not exist? That might be okay...

Reply:
Good callout!